

<a name=/exception>
<h2 byondver=508>exception</h2>

<dl><dt><b>See also:</b>
<dd><a href=#/proc/try>try and catch statements</a>
<dd><a href=#/world/proc/Error>Error proc (world)</a>
<dd><a href=#/proc/throw>throw statement</a>
<dd><a href=#/proc/EXCEPTION>EXCEPTION proc</a>
<dd><a href=#/proc/var/caller>caller var (proc)</a>
<dd><a href=#/{{appendix}}/stddef%2edm>stddef.dm file</a>
</dl>

<dl><dt><b>Vars:</b>
<dd>name: A text string (such as an error message) or other value
<dd>file: The filename where the error occurred, if debugging info is present
<dd>line: The line where the error occurred, if debugging info is present
<dd>desc: Detailed error info including call stack, only used when sent to world.Error()
</dl>

<p>This datum is created automatically when a runtime error is encountered,
<b>if</b> it happens within a try/catch block or you have defined a global
error handler with <tt>world.Error()</tt>. (The New() proc is not called when
this happens.) This provides a convenient package for getting file and line
number info associated with an error.

<p>If you throw your own exceptions, you do not have to use this, but the
<tt>EXCEPTION</tt> macro is provided to easily create one with the current
file and line number.

<p>The <tt>desc</tt> value is only filled in when you have a world.Error()
handler and there is no try/catch handling this error. Just like when no
handler is present, less detail will be provided after multiple runtime errors
have occurred. This only exists as a convenience feature for logging errors if
you want to use something other than world.log.

